src/main.rs

```diff
@@ -9,12 +8,19 @@ pub mod file_management;
 mod operations;
 mod tests;
 use std::os::fd::AsRawFd;
+use thiserror::Error;
 use timeout_readwrite::TimeoutReadExt;
-mod memory_management;
 
 static MEM_MAX: usize = 1 << 16;
 static PC_START: u16 = 0x3000;
+
+pub enum MemoryMappedRegisters {
```

Comment:
Lets add documentation about what do MemoryMappedRegisters mean and their variants.